lib/serverengine/socket_manager_win.rb

```diff
@@ -0,0 +1,136 @@
+#
+# ServerEngine
+#
+# Copyright (C) 2012-2013 Sadayuki Furuhashi
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'socket'
+require 'timeout'
+require 'ipaddr'
+require 'drb/drb'
+require 'serverengine/winsock'
+
+module ServerEngine
+  module SocketManagerWin
+    class Client
+      def initialize(drb_uri)
+        @drb_uri = drb_uri
+      end
+
+      def get_tcp(bind, port)
+        sm_server = DRb::DRbObject.new_with_uri(@drb_uri)
+        clientd_pid = Process.pid
+        proto_pack = sm_server.get_tcp_proto(clientd_pid.to_s, bind, port)
+        proto_map = MessagePack.unpack(proto_pack)
```

Comment:
I think you can use Marshal.load and dump here to remove dependency to MessagePack. It sounds better as a library of ServerEngine so that applications won't cause version conflicts.